addons/product/models/product_product.py

```diff
@@ -436,6 +434,224 @@ def _onchange_uom_id(self):
+        # Search product.attribute.value or create it #
+        ###############################################
+
+        attribute_to_values = defaultdict(list)  # attribute_name => list[value_name]
+        for vals in with_import_values:
+            for value in vals['import_attribute_values'].split(','):
+                attribute_name, value_name = value.split(':', 1) if ':' in value else (None, value)
+                attribute_name = attribute_name.strip()
+                if not attribute_name:
+                    raise ValueError(self.env._("Unable to import products with attribute value without attribute name (defined as: attribute:value): %s", value))
+                attribute_to_values[attribute_name].append(value_name.strip())
+
+        pa_pav_records = {}  # (attribute_name, value_name) => Record<'product.attribute.value'>
+        pa_records = {}  # attribute_name => Record<'product.attribute'>
+        values_to_create = []
+        for attribute_name, value_names in attribute_to_values.items():
+            # search attribute values
+            domain = Domain('name', 'in', value_names) & Domain('attribute_id.name', '=', attribute_name)
+            for pav in PAV.search(domain):
+                pa_pav_records[attribute_name, pav.name] = pav
+                pa_records[attribute_name] = pav.attribute_id
+            # create missing attribute values
+            if attribute_name not in pa_records:
+                pa_records[attribute_name] = PA.create({
+                    'name': attribute_name,
+                    'create_variant': 'dynamic',
+                    'display_type': 'radio',
+                })
+
+            # prepare to create missing attribute values
+            missing_pav = [key for value_name in value_names if (key := (attribute_name, value_name)) not in pa_pav_records]
+            values_to_create.extend(missing_pav)
+            pa_pav_records.update({key: None for key in missing_pav})
+
+        if values_to_create:
+            # create missing attribute values
+            pav = PAV.create([
+                {'name': value_name, 'attribute_id': pa_records[attribute_name].id}
+                for attribute_name, value_name in values_to_create
+            ])
+            for rec in pav:
+                pa_pav_records[rec.attribute_id.name, rec.name] = rec
+
+        ########################################
+        # Search product.template or create it #
+        ########################################
+
+        product_templates = PT.search(
+            Domain('name', 'in', [vals['name'] for vals in with_import_values]) |
+            Domain('id', 'in', [vals.get('product_tmpl_id') for vals in with_import_values]))
+        id2template = dict(zip(product_templates.ids, product_templates))  # id => Record<'product.template'>
+        name2template = dict(zip(product_templates.mapped('name'), product_templates))  # template_name => Record<'product.template'>
```

Comment:
What if there are multiple templates with the same name? 🤔 
It doesn't seem reliable enough, is it really needed ?
If no template id is provided, is it an issue to create a new one on the fly?